cabal-install/src/Distribution/Client/CmdClean.hs

```diff
@@ -80,16 +87,19 @@ cleanAction CleanFlags{..} extraArgs _ = do
         mdistDirectory = flagToMaybe cleanDistDir
         mprojectFile   = flagToMaybe cleanProjectFile
 
-    unless (null extraArgs) $
-        die' verbosity $ "'clean' doesn't take any extra arguments: "
-                         ++ unwords extraArgs
+    -- assume all files passed are the names of scripts
+    notScripts <- filterM (fmap not . doesFileExist) extraArgs
```

Comment:
Maybe we should be more precise about checking if target is a script file. Not sure if it is feasible but running a raw main.hs file gives:

```
> cabal run .\main.hs
cabal-3.6.2.0.exe: Failed extracting script block: `{- cabal:` start marker
not found
```

soi guess the check lives somewhere in the codebase

Reply:
@jneira I'm slightly confused by if your taking about `clean` or `run` or both?
re: `clean` to handle that properly clean would have to resolve targets which is #7506 that people have been talking about.
re: `run` is a new bug I introduced or did it work like that that before?

Reply:
Actually the error message looks like a regression, running `cabal v2-run cabal-install/main/Main.hs` yields:
```
Resolving dependencies...
cabal: The run command can only run an executable as a whole, not files or
modules within them, but the target 'main/Main.hs' refers to the file
main/Main.hsin the executable cabal.
```

which makes total sense, but the current error message does not make sense at all for newcomers.

Reply:
to elaborate on precise checking: I didn't want to explicitly check if all passed args are scripts because it's possible they might refer to things that where built as scripts in the past and where either deleted or turned into normal source files and you still want to clean the cache for them. That's a bit of an edge case so it might not be worth the worse error reporting.

re error messages: The exact error messages cabal prints depends on the context. In this case I believe it's whether there is a .cabal file in the current directory or not. It should probably give the same error message when you try to run the Main.hs of a package regardless of the directory your in, but this is the same behavior as before the pr, not a regression.